### PR TITLE
add as_psgi and new_psgi methods

### PR DESCRIPTION
**Pull request by rjbs**
These commits add methods to make it easier to use a Mason handler inline when building with Plack::Builder.

### CODE REVIEW

File: lib/HTML/Mason/PSGIHandler.pm [hunk 1]
```diff
@@ -120,6 +130,23 @@ HTML::Mason::PSGIHandler is a PSGI handler for HTML::Mason. It's based
...
 any web servers that support PSGI.
 
+=head1 METHODS
+
+=head1 as_psgi
+
+  my $app = $h->as_psgi;
+
+This method returns a subroutine to be used as a PSGI application, exactly as
+shown in the SYNOPSIS.
```
Comment (abh) [inline lib/HTML/Mason/PSGIHandler.pm]: "exactly as shown in the SYNOPSIS" makes it seem like there should be an example in the synopsis using as_psgi.  :-)

**Comment by rjbs** [comment]
I meant to imply that the subroutine returned is exactly like the subroutine built in the synopsis.  I can reword that, but maybe you'll have a way to say it more clearly, since I'm obviously already failing. ;-D

**Comment by abh** [comment]
I'd suggest just making an explicit example instead of referencing the SYNOPSIS.  :-)  Or make an explicit example in the SYNOPSIS.

**Comment by rjbs** [comment]
done!

**Comment by abh** [comment]
Cool, thanks!

**Pull request merged**